src/Admin/AbstractAdmin.php

```diff
@@ -2657,7 +2657,7 @@ public function hasAccess($action, $object = null)
     }
 
     /**
-     * @param string      $action
+     * @param mixed      $action
```

Comment:
:-1: this code was written at a time when type declarations were not available. Docblocks allowed and still allow us to establish a contract with the user. They cannot expect things to go well if they don't use a string here, and we won't change that.

Reply:
No they shouldn't. We explicitely documented that they can only expect `string` to produce results that have are supported and have been tested. On next major, we will very probably use `string` as a type hint here. Whether we are feeding it to `in_array` is irrelevant.